### PR TITLE
Update cipher suites

### PR DESCRIPTION
**Pull request by EvgenyAgafonchikov**
Sync May branch with https://github.com/Azure/azure-rest-api-specs/pull/6212

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by MikhailTryakhov** [comment]
@EvgenyAgafonchikov can you please confirm what api version issue is caused by a breaking change only so could be ignored

**Comment by EvgenyAgafonchikov** [comment]
@MikhailTryakhov, yes, NetworkValidation fails on Azure PS build step because of replacing of InterfaceEndpoints with PrivateEndpoints and we don't have the related update merged in Azure PS yet.

**Review by tjprescott — APPROVED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/applicationGateway.json [hunk 1]
```diff
@@ -2604,6 +2604,8 @@
       "type": "string",
       "description": "Ssl cipher suites enums.",
       "enum": [
+        "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/applicationGateway.json]: I agree with @sergey-shandar, let's add new enum values at the end.

**Review by anton-evseev — CHANGES_REQUESTED** [review]

**Comment by number213** [comment]
Addressed Sergey's comments from related PR https://github.com/Azure/azure-rest-api-specs/pull/6212
NetworkValidation's failure is a known issue with PowerShell, please ignore

**Comment by tjprescott** [comment]
@number213 are you requested changes implemented then? Your review is blocking the merge.

**Review by anton-evseev — APPROVED** [review]

**Pull request merged**